todo-list-app/src/app/todos/todos-list/todos-list.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Todos } from '../../todos.model';
+import { TodosService } from '../../todos.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-todos-list',
+  templateUrl: './todos-list.component.html',
+  styleUrls: ['./todos-list.component.css']
+})
+export class TodosListComponent implements OnInit, OnDestroy {
+  tasks: Todos[];
+  private todoSub: Subscription;
+  constructor(public todosService: TodosService) { }
+
+  ngOnInit() {
+    this.todosService.getTodos();
+    this.todoSub = this.todosService
+    .getTodosListener()
+    .subscribe((todosList: {todosList: Todos[], searchTodos: string}) => {
+      if (!todosList.searchTodos) {
+        this.tasks = todosList.todosList;
+      } else {
+        this.tasks = todosList.todosList.filter(todo => todo.title.toLowerCase().includes(todosList.searchTodos));
+      }
+    });
```

Comment:
Why subscribing to todolistener here again? It seems you have already updated todos in todoservice in subscribe callback